### PR TITLE
Build Web Workers ESM

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Web Workers ES modules are required to build easily  #2639 and are now supported by all browsers.

# Changes

- active ESM for Web Workers
- drop explicit usage of agent-js CJS

### CODE REVIEW

File: frontend/vite.config.ts [hunk 1]
```diff
@@ -109,6 +109,9 @@ const config: UserConfig = {
       },
     },
   },
+  worker: {
+    format: "es",
```
Comment (lmuntaner) [inline frontend/vite.config.ts:113]: Is this related to the other changes?

**Review by lmuntaner — APPROVED** [review]
LGTM, just one question.

File: frontend/vite.config.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/vite.config.ts:113]: That's the Vite flag that enables ES modules for web workers.
https://vitejs.dev/config/worker-options.html#worker-format

Comment (lmuntaner) [reply frontend/vite.config.ts:113]: Thanks to this we don't need to cjs explicit import?

Comment (peterpeterparker) [reply frontend/vite.config.ts:113]: Not anymore following this PR.

Comment (lmuntaner) [reply frontend/vite.config.ts:113]: Ok, thanks for the explanation
